subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidatePlugins.java

```diff
@@ -126,6 +134,12 @@ private static CharSequence toMessageList(List<String> problemMessages) {
         return builder;
     }
 
+    private Provider<JavaLauncher> toolchainLauncher() {
+        JavaToolchainSpec toolchain = getProject().getExtensions().getByType(JavaPluginExtension.class).getToolchain();
+        JavaToolchainService service = getProject().getExtensions().getByType(JavaToolchainService.class);
+        return service.launcherFor(toolchain);
+    }
+
```

Comment:
:x: This is missing the fact that the toolchain is an input to the task and is not compatible with the configuration cache because it uses the project at execution time.

You should add a `@Nested Property<JavaLauncher>` input to the task and set it from the plugin instead.